src/backend/distributed/commands/utility_hook.c

```diff
@@ -692,7 +711,9 @@ ProcessUtilityInternal(PlannedStmt *pstmt,
 	 */
 	if (EnableDDLPropagation)
 	{
-		if (ops && ops->postprocess)
+		DistObjectCurrentState = DISTOBJECTSTATE_POSTPROCESS;
+
+		if (ops && ops->postprocess && isObjectValid)
```

Comment:
There is 1 case with 'alter domain xxx drop constraint yyy'. At start, we validate the object calling its address callback. If the constraint is not valid, then it is ok, we are not processing preprocess and postprocess steps. Otherwise, we process those steps. The same address function is called in postprocess step. We end up with an invalid object returned from address callback by mistake because the constraint is already dropped before postprocess. Thus, I need to differentiate those steps. we have just 1 case which poses a problem because we use the same address function in preprocess, postprocess steps as in validation. An alternative solution would be to separate the validation logic from returning address if we want to remove those state enum. Another option would be to use 2 address methods for alterdomain in which we currently call GetObjectAddressFromParseTree in preprocess and postprocess steps. We can have 2 address methods, one of which is used in preprocess and postprocess steps and the other one is used only for validation. If you have other options, we can discuss them as well.

Reply:
> . The same address function is called in postprocess step

I think this should be a generic problem.

Should we do the following change such that we do not care about `ALlTER .. DROP ` cases? Would be good to have a separate PR if that solves your problem

```
diff --git a/src/backend/distributed/commands/common.c b/src/backend/distributed/commands/common.c
index 1c6a71de3..71d53f0ac 100644
--- a/src/backend/distributed/commands/common.c
+++ b/src/backend/distributed/commands/common.c
@@ -156,7 +156,7 @@ PostprocessAlterDistributedObjectStmt(Node *stmt, const char *queryString)
        const DistributeObjectOps *ops = GetDistributeObjectOps(stmt);
        Assert(ops != NULL);
 
-       ObjectAddress address = GetObjectAddressFromParseTree(stmt, false);
+       ObjectAddress address = GetObjectAddressFromParseTree(stmt, true);
        if (!ShouldPropagateObject(&address))
        {
                return NIL;

```